cpp/src/barretenberg/stdlib/recursion/verifier/verifier.hpp

```diff
@@ -325,18 +343,14 @@ aggregation_state<Curve> verify_proof(typename Curve::Composer* context,
         rhs_scalars.push_back(random_separator);
     }
 
-    // Check if recursive proof information is correctly set.
-    key->contains_recursive_proof.assert_equal(key->base_key->contains_recursive_proof,
```

Comment:
⚠️ it looks like this logic was moved into `from_field_pt_vector`, but if we use the `from_witness` of a verification constructor of a verification key we will be missing this constraint!

Reply:
@zac-williamson This is the constraint that was removed

Reply:
In the new `from_field_pt_vector`, `key->contains_recursive_proof` is a circuit constant, it's not a witness (it's value is constructed out of a regular `bool` type)

The constraint is removed because there is no longer a `contains_recursive_proof` witness value at all - we effectively generate 2 different circuits depending on whether the input into `from_field_pt_vector` describes a recursive circuit or not.